d1.go

```diff
@@ -0,0 +1,197 @@
+)
+
+type D1Database struct {
+	Name      string     `json:"name"`
+	NumTables int        `json:"num_tables"`
+	UUID      string     `json:"uuid"`
+	Version   string     `json:"version"`
+	CreatedAt *time.Time `json:"created_at"`
+	FileSize  int64      `json:"file_size"`
+}
+
+type ListD1Parameters struct {
+	Name string `url:"name,omitempty"`
+	ResultInfo
+}
+
+type ListD1Response struct {
+	Result []D1Database `json:"result"`
+	Response
+	ResultInfo `json:"result_info"`
+}
+
+type CreateD1Parameters struct {
+	Name string `json:"name"`
+}
+
+type D1DatabaseResponse struct {
+	Result D1Database `json:"result"`
+	Response
+}
+
+type QueryD1Parameters struct {
+	DatabaseID string   `json:"database_id"`
+	SQL        string   `json:"sql"`
+	Parameters []string `json:"params"`
+}
+
+type D1DatabaseMetadata struct {
+	ChangedDB   bool    `json:"changed_db"`
+	Changes     int     `json:"changes"`
+	Duration    float64 `json:"duration"`
+	LastRowID   int     `json:"last_row_id"`
+	RowsRead    int     `json:"rows_read"`
+	RowsWritten int     `json:"rows_written"`
+	SizeAfter   int     `json:"size_after"`
+}
+
+type D1Result struct {
+	Success bool               `json:"success"`
+	Results []map[string]any   `json:"results"`
+	Meta    D1DatabaseMetadata `json:"meta"`
+}
+type QueryD1Response struct {
+	Result []D1Result `json:"result"`
+	Response
+}
+
+// ListD1Databases returns all databases for an account.
+//
+// API reference: https://developers.cloudflare.com/api/operations/cloudflare-d1-list-databases
+func (api *API) ListD1Databases(ctx context.Context, rc *ResourceContainer, params ListD1Parameters) ([]D1Database, *ResultInfo, error) {
+	if rc.Identifier == "" {
+		return []D1Database{}, &ResultInfo{}, ErrMissingAccountID
+	}
+	baseURL := fmt.Sprintf("/accounts/%s/d1/database", rc.Identifier)
+	autoPaginate := true
+	if params.PerPage >= 1 || params.Page >= 1 {
+		autoPaginate = false
+	}
+
+	if params.PerPage < 1 {
+		params.PerPage = 100
+	}
+
+	if params.Page < 1 {
+		params.Page = 1
+	}
+	var databases []D1Database
+	var r ListD1Response
+	for {
+		uri := buildURI(baseURL, params)
+		res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+		if err != nil {
+			return []D1Database{}, &ResultInfo{}, fmt.Errorf("%s: %w", errMakeRequestError, err)
+		}
+
+		err = json.Unmarshal(res, &r)
+		if err != nil {
+			return []D1Database{}, &ResultInfo{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+		}
+		databases = append(databases, r.Result...)
+		params.ResultInfo = r.ResultInfo.Next()
+		if params.ResultInfo.Done() || !autoPaginate {
+			break
+		}
+	}
+	return databases, &r.ResultInfo, nil
+}
+
+// CreateD1Database creates a new database for an account.
+//
+// API reference: https://developers.cloudflare.com/api/operations/cloudflare-d1-create-database
+func (api *API) CreateD1Database(ctx context.Context, rc *ResourceContainer, params CreateD1Parameters) (D1Database, error) {
```

Comment:
the third param should be `<Method>Params`

```suggestion
func (api *API) CreateD1Database(ctx context.Context, rc *ResourceContainer, params CreateD1DatabaseParams) (D1Database, error) {
```